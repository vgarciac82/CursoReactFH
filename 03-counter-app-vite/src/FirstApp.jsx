import PropTypes from "prop-types";

const FirstApp = ({ title, subtitle = "REACT DEVELOPER" }) => {
  const getResult = (a, b) => {
    return a + b;
  };

  return (
    <>
      
        <h1><div data-testid="test-title">{title}</div></h1>
      
      <h2>{subtitle}</h2>
      <p>My first React App</p>
    </>
  );
};

export default FirstApp;

FirstApp.proppTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  subtitle: "BEST REACT DEVELOPER",
};
