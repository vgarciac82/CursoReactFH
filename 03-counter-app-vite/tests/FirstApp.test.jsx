import { render } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe("Pruebas en <FistApp />", () => {
  test("Debe hacer match con el primer snapshot", () => {
    const title = "Hola Mundo";
    const { container } = render(<FirstApp title="{title}" />);

    expect(container).toMatchSnapshot();
    
  });

  test("Debe hacer match con el primer snapshot", () => {
    const title = "Hola Mundo";
    const subtitle = "BEST REACT DEVELOPER";

    const { container, getByText, getByTestId } = render(<FirstApp title={title} />);

    expect(getByText(title)).toBeTruthy();

    const h1 = container.querySelector("h1");
    expect(h1.innerHTML).toContain(title);
   
    expect( getByTestId('test-title').innerHTML ).toBe(title);
  });
});
