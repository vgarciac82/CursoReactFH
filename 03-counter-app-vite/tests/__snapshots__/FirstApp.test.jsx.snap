// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FistApp /> Debe hacer match con el primer snapshot 1`] = `
<div>
  <h1>
    <div
      data-testid="test-title"
    >
      {title}
    </div>
  </h1>
  <h2>
    BEST REACT DEVELOPER
  </h2>
  <p>
    My first React App
  </p>
</div>
`;
